### PR TITLE
Implement registration of per-package build logs

### PR DESCRIPTION
**Pull request by mrkkrp**

### CODE REVIEW

**Comment by facundominguez** [comment]
I'm missing some context. What does this PR implement? What is **the registration** of a package build log?

**Comment by mrkkrp** [comment]
OK, this is a preparation step for implementing generation of static site that we could then serve and provide info about Stackage HEAD builds in a readable form for the community.

So if we want to generate a static site, it makes sense to try to re-generate it completely on every run because otherwise we're asking for trouble:

* What if we discover some sort of issue, e.g. with how pages are hyperlinked? We can't easily repair that if we generate only new pages.
* If we change something in layout or content of pages, we would like these changes be applied to all content that we serve.

In short, there is no reason not to regenerate everything, it's not a bottleneck and it's not resource-intensive.

So if we want to regenerate the whole site, we need all inputs for this. All inputs include inputs for older runs too (because we want to show a history of builds and diffs between them on the site).

Build results are stored already and we have a history file that can be truncated over time, etc. But this is not so for per-build logs which are stored in a separate directory by Stackage curator (i.e. logs that contain compilation errors and other useful information about how a build went). Before this PR the logs were only available for the build that we performed in this CircleCI run, now they can be "registered" so they are copied and persisted and cleaned up automatically along with other info. And so they can be used for generation of site content.

File: .circleci/config.yml [hunk 1]
```diff
@@ -37,11 +37,8 @@ jobs:
             stackage-curator make-bundle --allow-newer --jobs $JOBS --plan-file $BUILD_PLAN.yaml --docmap-file docmap-file.yaml --bundle-file bundle-file.yaml --target $BUILD_PLAN --skip-haddock --skip-hoogle --skip-benches --no-rebuild-cabal -v 2>&1 | tee build.log  || true
             stack setup
             stack build
-            stack exec stackage-head -- add --ghc-metadata metadata.json --build-log build.log --target $BUILD_PLAN --outdir $HOME/build-reports
-            stack exec stackage-head -- truncate --history-length 10 --outdir $HOME/build-reports
-      - store_artifacts:
-          path: logs
-          destination: logs
```
Comment (facundominguez) [inline .circleci/config.yml]: How are the logs persisted now that they are not stored as artifacts?

**Comment by facundominguez** [comment]
LGTM

What is a `HistoryItem`?

**Pull request merged**